RemoteLayerTreeDrawingAreaProxy needs to be made more robustly aware of multiple processes.

mattwoodrow opened the pull request:
#### db9def36c07cf523e41e48ceb18b575c1e5c3399
<pre>
RemoteLayerTreeDrawingAreaProxy needs to be made more robustly aware of multiple processes.
<a href="https://bugs.webkit.org/show_bug.cgi?id=279804">https://bugs.webkit.org/show_bug.cgi?id=279804</a>
&lt;<a href="https://rdar.apple.com/134885033">rdar://134885033</a>&gt;

Reviewed by Alex Christensen.

Added &apos;MainFrame&apos; to some getters to make it clearer that they return state
specific to the main frame.

Removed subframe support from waitForDidUpdateActivityState, since we only ever
call it for the main frame (and we only ever want to support blocking the UI
process for the main frame).

Added better support for animations with remote subframes. Timeline origin and
current time moved to be per-process data. RemoteLayerTreeNode now stores the
ProcessIdentifier for the process that created it.
RemoteLayerTreeDrawingAreaProxy animation functions now take the
ProcessIdentifier, and return/send data for the right process.

lastVisibleTransactionID/transactionIDForPendingCACommit/activityStateChangeID
moved out of the per-process state, since these are only relevant to the main
frame.

setPreferredFramesPerSecond changed to ignore requests from remote subframes.
This should match the behaviour without site-isolation enabled. We have the
opportunity to do better (and truly throttle subframes at the process boundary),
but it&apos;s a bigger piece of work.

deviceScaleFactorDidChange and colorSpaceDidChange now get forwarded to all
processes, not just that of the main frame.

Some code that tried to read subframe RemoteLayerTreeTransactions and use the
data to configure the root now skipped (like
RemoteLayerTreeDrawingAreaProxyMac::didCommitLayerTree).

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _didCommitLoadForMainFrame]):
(-[WKWebView _restorePageScrollPosition:scrollOrigin:previousObscuredInset:scale:]):
(-[WKWebView _restorePageStateToUnobscuredCenter:scale:]):
* Source/WebKit/UIProcess/DrawingAreaProxy.h:
(WebKit::DrawingAreaProxy::waitForDidUpdateActivityState):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
(WebKit::RemoteLayerTreeDrawingAreaProxy::nextMainFrameLayerTreeTransactionID const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::lastCommittedMainFrameLayerTreeTransactionID const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::setPreferredFramesPerSecond):
(WebKit::RemoteLayerTreeDrawingAreaProxy::nextLayerTreeTransactionID const): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxy::lastCommittedLayerTreeTransactionID const): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxy::acceleratedTimelineTimeOrigin const): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxy::animationCurrentTime const): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::deviceScaleFactorDidChange):
(WebKit::RemoteLayerTreeDrawingAreaProxy::processStateForConnection):
(WebKit::RemoteLayerTreeDrawingAreaProxy::processStateForIdentifier):
(WebKit::RemoteLayerTreeDrawingAreaProxy::processStateForIdentifier const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::connectionForIdentifier):
(WebKit::RemoteLayerTreeDrawingAreaProxy::forEachProcessState):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeTransaction):
(WebKit::RemoteLayerTreeDrawingAreaProxy::acceleratedAnimationDidStart):
(WebKit::RemoteLayerTreeDrawingAreaProxy::acceleratedAnimationDidEnd):
(WebKit::RemoteLayerTreeDrawingAreaProxy::didRefreshDisplay):
(WebKit::RemoteLayerTreeDrawingAreaProxy::waitForDidUpdateActivityState):
(WebKit::RemoteLayerTreeDrawingAreaProxy::acceleratedTimelineTimeOrigin const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::animationCurrentTime const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::animationDidStart):
(WebKit::RemoteLayerTreeHost::animationDidEnd):
(WebKit::RemoteLayerTreeHost::acceleratedTimelineTimeOrigin const):
(WebKit::RemoteLayerTreeHost::animationCurrentTime const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.mm:
(WebKit::RemoteLayerTreeNode::RemoteLayerTreeNode):
(WebKit::RemoteLayerTreeNode::setAcceleratedEffectsAndBaseValues):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::setPreferredFramesPerSecond):
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.h:
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::didCommitLayerTree):
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::setPreferredFramesPerSecond):
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::colorSpaceDidChange):
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::updateZoomTransactionID):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::waitForDidUpdateActivityState):
(WebKit::WebPageProxy::didCommitLoadForFrame):
* Source/WebKit/UIProcess/ios/WKContentView.mm:
(-[WKContentView didUpdateVisibleRect:unobscuredRect:contentInsets:unobscuredRectInScrollViewCoordinates:obscuredInsets:unobscuredSafeAreaInsets:inputViewBounds:scale:minimumScale:viewStability:enclosedInScrollableAncestorView:sendEvenIfUnchanged:]):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _touchEventsRecognized:]):
(-[WKContentView mouseInteraction:changedWithEvent:]):
(-[WKContentView _simulateElementAction:atLocation:]):
* Source/WebKit/UIProcess/mac/TiledCoreAnimationDrawingAreaProxy.h:
* Source/WebKit/UIProcess/mac/TiledCoreAnimationDrawingAreaProxy.mm:
(WebKit::TiledCoreAnimationDrawingAreaProxy::waitForDidUpdateActivityState):

Canonical link: <a href="https://commits.webkit.org/284515@main">https://commits.webkit.org/284515@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/086c91d4235c50da936cf163491f956f8a384a3b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69570 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48970 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22260 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73655 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20728 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56771 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20579 "Built successfully") | [![loading-orange](https://github-production-user-asset-6210df.s3.amazonaws.com/3098702/291015173-08c448be-ac0a-4fd6-92a3-8165057445b7.png) 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55303 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running run-layout-tests-in-stress-mode; 22 flakes 75 failures; Uploaded test results; 10 flakes 61 failures; Compiled WebKit (warnings); 1 flakes 62 failures; Running analyze-layout-tests-results") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13766 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72636 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44658 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/60059 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35782 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41324 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/19105 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63259 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17835 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75365 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/13552 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/17051 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/62970 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13592 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/60142 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62884 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15479 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10910 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4535 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44774 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45848 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/47043 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45589 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
```diff
@@ -68,12 +68,11 @@ class RemoteLayerTreeDrawingAreaProxy : public DrawingAreaProxy {
 
-    void acceleratedAnimationDidStart(WebCore::PlatformLayerIdentifier, const String& key, MonotonicTime startTime);
-    void acceleratedAnimationDidEnd(WebCore::PlatformLayerIdentifier, const String& key);
+    void acceleratedAnimationDidStart(WebCore::PlatformLayerIdentifier, WebCore::ProcessIdentifier, const String& key, MonotonicTime startTime);
```
Isn't the PlatformLayerIdentifier process-qualified already?

mattwoodrow replied:
So it is! I missed that, will fix.

achristensen07 commented on Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
```diff
@@ -1430,7 +1430,7 @@ - (void)_restorePageStateToUnobscuredCenter:(std::optional<WebCore::FloatPoint>)
         return;
 
-    _perProcessState.firstTransactionIDAfterPageRestore = downcast<WebKit::RemoteLayerTreeDrawingAreaProxy>(*_page->drawingArea()).nextLayerTreeTransactionID();
+    _perProcessState.firstTransactionIDAfterPageRestore = downcast<WebKit::RemoteLayerTreeDrawingAreaProxy>(*_page->drawingArea()).nextMainFrameLayerTreeTransactionID();
```
Should we call this nextMainFrameProcessLayerTreeTransactionID?  A transaction comes from all the frames in a process, and if we have a.com with an iframe at b.com with a grandchild iframe at a.com, then both main frame and grandchild frame should come in the same transaction, right?

mattwoodrow replied:
Naming things is hard :)

We have an IPC message called `commitLayerTree`, which has a `Vector` of `RemoteLayerTreeTransaction`s (one per root frame in the sending process).

I've been thinking of 'transaction' as the per-root-frame thing, and the per-process outer thing doesn't really have an explicit name (though is definitely transactional in nature). Transaction-set ?

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
```diff
@@ -100,6 +99,28 @@ class RemoteLayerTreeDrawingAreaProxy : public DrawingAreaProxy {
+    // with a commitLayerTree message (ideally before the next
+    // displayDidRefresh, otherwise we mark it as missed and send
+    // it when commitLayerTree does arrive).
+    enum CommitLayerTreeMessageState { CommitLayerTreePending, NeedsDisplayDidRefresh, MissedCommit, Idle };
```
Can we make this an enum class?  c-style enums are less preferred in our code, but in this case it's a nested enum so the names are scoped so it's not as big of an issue.  Also, you're just moving this so probably not something for this PR.

mattwoodrow replied:
Indeed, that would be a good cleanup.

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.mm:
```diff
@@ -211,8 +212,11 @@ - (void)didChangeNominalFramesPerSecond
-void RemoteLayerTreeDrawingAreaProxyIOS::setPreferredFramesPerSecond(FramesPerSecond preferredFramesPerSecond)
+void RemoteLayerTreeDrawingAreaProxyIOS::setPreferredFramesPerSecond(IPC::Connection& connection, FramesPerSecond preferredFramesPerSecond)
 {
+    if (!m_webProcessProxy->hasConnection() || &m_webProcessProxy->connection() != &connection)
```
AuxiliaryProcessProxy::hasConnection(const IPC::Connection&) does this in one call.
Also, aren't we working towards removing m_webProcessProxy?

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
```diff
@@ -667,6 +710,19 @@
+
+MonotonicTime RemoteLayerTreeDrawingAreaProxy::animationCurrentTime(WebCore::ProcessIdentifier processIdentifier) const
+{
+    auto& state =  processStateForIdentifier(processIdentifier);
```
extra space.  Same on line 716

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
```diff
@@ -177,7 +177,7 @@
     }
 
-    ASSERT(m_webProcessProxy->hasConnection() && &connection == &m_webProcessProxy->connection());
+    RELEASE_ASSERT(m_webProcessProxy->hasConnection() && &connection == &m_webProcessProxy->connection());
```
This can use the AuxiliaryProcessProxy::hasConnection that takes a connection parameter to simplify.

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
```diff
@@ -188,6 +188,42 @@
+    return m_webPageProxyProcessState;
+}
+
+const RemoteLayerTreeDrawingAreaProxy::ProcessState& RemoteLayerTreeDrawingAreaProxy::processStateForIdentifier(WebCore::ProcessIdentifier identifier) const
```
This is duplicate code with the non-const version.  Not ideal.

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm:
```diff
@@ -541,7 +550,7 @@ explicit RemoteLayerTreeDisplayLinkClient(WebPageProxyIdentifier pageID)
-    m_transactionIDAfterEndingTransientZoom = nextLayerTreeTransactionID();
+    m_transactionIDAfterEndingTransientZoom = nextMainFrameLayerTreeTransactionID();
 }
 
```
Nit: could you remove the extra space here too?

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
```diff
@@ -188,6 +188,42 @@
+RemoteLayerTreeDrawingAreaProxy::ProcessState& RemoteLayerTreeDrawingAreaProxy::processStateForIdentifier(WebCore::ProcessIdentifier identifier)
+{
+    RefPtr webProcess = WebProcessProxy::processForIdentifier(identifier);
+    RELEASE_ASSERT(webProcess);
```
I think it would be better to not look up the webProcess at all.  We can just call pair.key.process().coreProcessIdentifier() and compare it with identifier below.  Same with the checking that it is the web page proxy's identifier.

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
```diff
@@ -91,6 +90,28 @@ class RemoteLayerTreeDrawingAreaProxy : public DrawingAreaProxy {
+        TransactionID pendingLayerTreeTransactionID;
+
+#if ENABLE(THREADED_ANIMATION_RESOLUTION)
+        Seconds m_acceleratedTimelineTimeOrigin;
```
This is a struct.  These should probably not have the m_ prefix like the ones above don't have.

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
```diff
@@ -188,6 +188,42 @@
+    RefPtr webProcess = WebProcessProxy::processForIdentifier(identifier);
+    RELEASE_ASSERT(webProcess);
+
+    for (auto pair : m_remotePageProcessState) {
```
Ideally we wouldn't be iterating through a HashMap like this.  Maybe we should keep a parallel HashMap of process identifier to WeakPtr of the state or something.

mattwoodrow replied:
This is sort of an existing problem, with looking up the process state using an `IPC::Connection` for all incoming messages.

It's a bit unfortunate that we go through the `RemotePageDrawingAreaProxy`'s `didReceiveMessage` function (which knows the process identifier and process proxy), through to `RemoteLayerTreeDrawingAreaProxy`'s receiver functions, which then have to reconstruct this info using the connection.

Maybe we should have the `RemotePageDrawingAreaProxy` do the lookup and store it (as the 'active' state or something). Sort of sad statefulness to have, but should work better.

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
```diff
@@ -172,20 +172,43 @@
+    for (auto& [key, value] : m_remotePageProcessState) {
+        RefPtr webProcess = WebProcessProxy::processForIdentifier(key);
+        RELEASE_ASSERT(webProcess);
+        if (webProcess->hasConnection(connection))
```
I think this should be if (webProcess && webProcess->hasConnection(...))
A web process other than the one we received a message from may be being torn down.

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
```diff
@@ -172,20 +172,43 @@
-        callback(value, key.protectedProcess());
+    for (auto& [key, value] : m_remotePageProcessState) {
+        RefPtr webProcess = WebProcessProxy::processForIdentifier(key);
+        RELEASE_ASSERT(webProcess);
```
Ditto, this could be a null check and a debug assertion.

mattwoodrow committed:
RemoteLayerTreeDrawingAreaProxy needs to be made more robustly aware of multiple processes.
https://bugs.webkit.org/show_bug.cgi?id=279804
<rdar://134885033>

Reviewed by Alex Christensen.

Added 'MainFrame' to some getters to make it clearer that they return state
specific to the main frame.

Removed subframe support from waitForDidUpdateActivityState, since we only ever
call it for the main frame (and we only ever want to support blocking the UI
process for the main frame).

Added better support for animations with remote subframes. Timeline origin and
current time moved to be per-process data. RemoteLayerTreeNode now stores the
ProcessIdentifier for the process that created it.
RemoteLayerTreeDrawingAreaProxy animation functions now take the
ProcessIdentifier, and return/send data for the right process.

lastVisibleTransactionID/transactionIDForPendingCACommit/activityStateChangeID
moved out of the per-process state, since these are only relevant to the main
frame.

setPreferredFramesPerSecond changed to ignore requests from remote subframes.
This should match the behaviour without site-isolation enabled. We have the
opportunity to do better (and truly throttle subframes at the process boundary),
but it's a bigger piece of work.

deviceScaleFactorDidChange and colorSpaceDidChange now get forwarded to all
processes, not just that of the main frame.

Some code that tried to read subframe RemoteLayerTreeTransactions and use the
data to configure the root now skipped (like
RemoteLayerTreeDrawingAreaProxyMac::didCommitLayerTree).

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _didCommitLoadForMainFrame]):
(-[WKWebView _restorePageScrollPosition:scrollOrigin:previousObscuredInset:scale:]):
(-[WKWebView _restorePageStateToUnobscuredCenter:scale:]):
* Source/WebKit/UIProcess/DrawingAreaProxy.h:
(WebKit::DrawingAreaProxy::waitForDidUpdateActivityState):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
(WebKit::RemoteLayerTreeDrawingAreaProxy::nextMainFrameLayerTreeTransactionID const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::lastCommittedMainFrameLayerTreeTransactionID const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::setPreferredFramesPerSecond):
(WebKit::RemoteLayerTreeDrawingAreaProxy::nextLayerTreeTransactionID const): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxy::lastCommittedLayerTreeTransactionID const): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxy::acceleratedTimelineTimeOrigin const): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxy::animationCurrentTime const): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::deviceScaleFactorDidChange):
(WebKit::RemoteLayerTreeDrawingAreaProxy::processStateForConnection):
(WebKit::RemoteLayerTreeDrawingAreaProxy::processStateForIdentifier):
(WebKit::RemoteLayerTreeDrawingAreaProxy::processStateForIdentifier const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::connectionForIdentifier):
(WebKit::RemoteLayerTreeDrawingAreaProxy::forEachProcessState):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeTransaction):
(WebKit::RemoteLayerTreeDrawingAreaProxy::acceleratedAnimationDidStart):
(WebKit::RemoteLayerTreeDrawingAreaProxy::acceleratedAnimationDidEnd):
(WebKit::RemoteLayerTreeDrawingAreaProxy::didRefreshDisplay):
(WebKit::RemoteLayerTreeDrawingAreaProxy::waitForDidUpdateActivityState):
(WebKit::RemoteLayerTreeDrawingAreaProxy::acceleratedTimelineTimeOrigin const):
(WebKit::RemoteLayerTreeDrawingAreaProxy::animationCurrentTime const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::animationDidStart):
(WebKit::RemoteLayerTreeHost::animationDidEnd):
(WebKit::RemoteLayerTreeHost::acceleratedTimelineTimeOrigin const):
(WebKit::RemoteLayerTreeHost::animationCurrentTime const):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.mm:
(WebKit::RemoteLayerTreeNode::RemoteLayerTreeNode):
(WebKit::RemoteLayerTreeNode::setAcceleratedEffectsAndBaseValues):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::setPreferredFramesPerSecond):
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.h:
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::didCommitLayerTree):
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::setPreferredFramesPerSecond):
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::colorSpaceDidChange):
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::updateZoomTransactionID):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::waitForDidUpdateActivityState):
(WebKit::WebPageProxy::didCommitLoadForFrame):
* Source/WebKit/UIProcess/ios/WKContentView.mm:
(-[WKContentView didUpdateVisibleRect:unobscuredRect:contentInsets:unobscuredRectInScrollViewCoordinates:obscuredInsets:unobscuredSafeAreaInsets:inputViewBounds:scale:minimumScale:viewStability:enclosedInScrollableAncestorView:sendEvenIfUnchanged:]):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _touchEventsRecognized:]):
(-[WKContentView mouseInteraction:changedWithEvent:]):
(-[WKContentView _simulateElementAction:atLocation:]):
* Source/WebKit/UIProcess/mac/TiledCoreAnimationDrawingAreaProxy.h:
* Source/WebKit/UIProcess/mac/TiledCoreAnimationDrawingAreaProxy.mm:
(WebKit::TiledCoreAnimationDrawingAreaProxy::waitForDidUpdateActivityState):

Canonical link: https://commits.webkit.org/284515@main
